File: 1365-how-many-numbers-are-smaller-than-the-current-number/1365-how-many-numbers-are-smaller-than-the-current-number.cpp
class Solution {
public:
    vector<int> smallerNumbersThanCurrent(vector<int>& nums) {
        vector<int > val;
         int len = nums.size();
         int flag  = 0 ;
        for(int i= 0 ; i < len ;i++){
            flag = 0 ;
            for(int j = 0 ; j < len ; j++){
                if(nums[i] > nums[j])
                    flag++;
            }
            val.emplace_back(flag);
        }
        
        return val;
    }
};

File: 1365-how-many-numbers-are-smaller-than-the-current-number/1365-how-many-numbers-are-smaller-than-the-current-number.py
class Solution:
    def smallerNumbersThanCurrent(self, nums: List[int]) -> List[int]:
        
        counter = [0] * (max(nums)+1)
        result = [0] * len(nums)
        
        
        for i in range(len(nums)):  
            counter[nums[i]] +=1
            
        for i in range( 1 , len(counter)):
            counter[i] += counter[i-1];
            
        for i in range(len(result)):
            if nums[i] == 0:
                result[i] = 0
            else:
                result[i] = counter[nums[i]-1]
            
        return result
            
            